src/Adapter/Product/Update/RelatedProductsUpdater.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\Update;
+
+use PrestaShop\PrestaShop\Adapter\Product\Repository\ProductRepository;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+use PrestaShopException;
+use Product;
+
+/**
+ * Provides methods to update related products (a.k.a accessories)
+ */
+class RelatedProductsUpdater
+{
+    /**
+     * @var ProductRepository
+     */
+    private $productRepository;
+
+    /**
+     * @param ProductRepository $productRepository
+     */
+    public function __construct(
+        ProductRepository $productRepository
+    ) {
+        $this->productRepository = $productRepository;
+    }
+
+    /**
+     * @param Product $product
+     * @param ProductId[] $relatedProductIds
+     */
+    public function set(Product $product, array $relatedProductIds): void
```

Comment:
```suggestion
    public function setRelatedProducts(ProductId $productId, array $relatedProductIds): void
```

Why not rely on the productId here, like it was done in `ProductSupplierUpdater::setProductSuppliers` The updater already has a dependency to the repository anyway But it would make one less in the handler

Reply:
the setRelatedProducts method naming is kinda redundant as its clearly expressed in class name, but ok, it will be clearer.

Reply:
Yeah I agree with the redundant name, I thought so as well at first but since we used this for ProductSupplier I guess it's ok 🤷‍♂️